### PR TITLE
update verify shellcheck to reduce the time to less than 15min

### LINKED ISSUE #1849: verify job is taking >15 min
**What would you like to be cleaned**:

Verify is taking too long

https://testgrid.k8s.io/sig-scheduling#pull-kueue-verify-main&width=20

Would increasing the number of CPUs help? Otherwise, what is being particularly slow?

**Why is this needed**:

This adds more time to the release process.

### PR DESCRIPTION
**Pull request by Vandit1604**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
This PR addresses `verify` rule taking more than 15 mins in Prow test suite.

#### Which issue(s) this PR fixes:

Part of #1849 

#### Special notes for your reviewer:
For more info, see https://github.com/kubernetes-sigs/kueue/issues/1849#issuecomment-2002527593

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: hack/verify-shellcheck.sh [hunk 1]
```diff
@@ -17,18 +17,54 @@
...
+  "--external-sources"
+  # include our disabled lints
+  "--exclude=${SHELLCHECK_DISABLED}"
+  # set colorized output
+  "--color=${SHELLCHECK_COLORIZED_OUTPUT}"
+)
+
 # Currently disabled these errors will take care of them later
-DISABLED_ERRORS="SC2002,SC3028,SC3054,SC3014,SC3040,SC3046,SC3030,SC3010,SC3037,SC3045,SC3006,SC3018,SC3016,SC3011,SC3044,SC3043,SC3060,SC3024,SC1091,SC2066,SC2086,SC2034,SC1083,SC1009,SC1073,SC1072,SC2155,SC2046"
+SHELLCHECK_DISABLED="SC2002,SC3028,SC3054,SC3014,SC3040,SC3046,SC3030,SC3010,SC3037,SC3045,SC3006,SC3018,SC3016,SC3011,SC3044,SC3043,SC3060,SC3024,SC1091,SC2066,SC2086,SC2034,SC1083,SC1009,SC1073,SC1072,SC2155,SC2046"
```
Comment (alculquicondor) [inline hack/verify-shellcheck.sh:37]: shouldn't this variable be defined before line 25?

**Comment by alculquicondor** [comment]
while this is faster, verify is still taking more than 10 min

Maybe there are more things we can do in other verifications.

File: hack/verify-shellcheck.sh (hunk 1 above)
Comment (Vandit1604) [reply hack/verify-shellcheck.sh:37]: Fixed.

**Comment by Vandit1604** [comment]
Is there a way to check how much time a rule is taking? I tried adding `time` command to each rule's definition, but it failed with `golangci-lint`. I also checked in k/k's Makefile to see in which rule do they use shell check, but they don't use it in any of the rules. 

Furthermore, I'll check other verifications tomorrow,

File: hack/verify-shellcheck.sh [hunk 2]
```diff
@@ -17,18 +17,54 @@
...
+  # we need this because we specify one file at a time in order to trivially
+  # detect which files are failing
+  "--external-sources"
+  # include our disabled lints
+  "--exclude=${SHELLCHECK_DISABLED}"
+  # set colorized output
+  "--color=${SHELLCHECK_COLORIZED_OUTPUT}"
+)
+
 # Currently disabled these errors will take care of them later
```
Comment (alexandear) [inline hack/verify-shellcheck.sh:37]: Perhaps we should move this comment to line 23.

Comment (alexandear) [inline hack/verify-shellcheck.sh:37]: Perhaps we should move this comment to line 23.

**Pull request merged**